engine/src/test/java/io/zeebe/engine/processor/workflow/job/JobTimeOutTest.java

```diff
@@ -82,6 +82,29 @@ public void shouldTimeOutJob() {
             JobIntent.ACTIVATED);
   }
 
+  @Test
+  public void shouldTimeOutAfterReprocessing() {
+    // given
+    final long jobKey = ENGINE.createJob(jobType, PROCESS_ID).getKey();
+    final long timeout = 10L;
+
+    ENGINE.jobs().withType(jobType).withTimeout(timeout).activate();
+    ENGINE.increaseTime(JobTimeoutTrigger.TIME_OUT_POLLING_INTERVAL);
+    jobRecords(TIME_OUT).withRecordKey(jobKey).getFirst();
```

Comment:
Do you need two jobs for this test case? I don't see what is the first job used for.

Reply:
As discussed only with the second job we see the real problem, that we can't timeout further jobs